File: teamscale-client/src/main/java/com/teamscale/client/ProcessInformation.java
package com.teamscale.client;

/** Information about the process and machine the profiler is attached to. */
public class ProcessInformation {

	/** Hostname of the machine it is running on */
	public final String hostname;

	/** Profiled PID */
	public final String pid;

	/** The timestamp at which the process was started. */
	public final long startedAtTimestamp;

	public ProcessInformation(String hostname, String pid, long startedAtTimestamp) {
		this.hostname = hostname;
		this.pid = pid;
		this.startedAtTimestamp = startedAtTimestamp;
	}
}

File: teamscale-client/src/main/kotlin/com/teamscale/client/ProcessInformation.kt
package com.teamscale.client

/** Information about the process and machine the profiler is attached to.  */
class ProcessInformation(
	/** Hostname of the machine it is running on  */
	val hostname: String,
	/** Profiled PID  */
	val pid: String,
	/** The timestamp at which the process was started.  */
	val startedAtTimestamp: Long
)
